api/src/main/java/org/openmrs/module/htmlformentry/widget/ZonedDateTimeWidget.java

```
@@ -26,29 +28,59 @@ public ZonedDateTimeWidget() {
-			} else {
-				return WidgetFactory.displayEmptyValue(timeWidget.getHideSeconds() ? "___:___" : "___:___:___");
+			boolean timezoneConversions = BooleanUtils.toBoolean(
+			    Context.getAdministrationService().getGlobalProperty(HtmlFormEntryConstants.GP_TIMEZONE_CONVERSIONS));
+			if (BooleanUtils.isNotTrue(timezoneConversions)) {
+				//The server and client are on the same timezone
+				return WidgetFactory.displayValue(datetimeFormat().format(initialValue));
+			}
+			//If Timezone.Conversions is true but the UP with client timezone is empty, it shows an error, because we dont know the client timezone.
+			if (BooleanUtils.isTrue(timezoneConversions) && StringUtils.isEmpty(this.getClientTimezone())) {
+				throw new RuntimeException(
+				        Context.getMessageSourceService().getMessage("htmlformentry.error.emptyClientTimezoneUserProperty"));
 			}
+			return WidgetFactory.displayValue(toClientTimezone(initialValue,
+			    Context.getAdministrationService().getGlobalProperty(HtmlFormEntryConstants.FORMATTER_DATETIME_NAME)));
 		} else {
 			StringBuilder sb = new StringBuilder();
 			
 			// the date part
 			sb.append(super.generateHtml(context));
 			
 			// the time part
-			Calendar valAsCal = initialValue != null ? toUTCCalendar(initialValue) : null;
+			Calendar valAsCal = Calendar.getInstance();
+			if (initialValue != null) {
+				SimpleDateFormat sdf = new SimpleDateFormat(Context.getAdministrationService().getGlobalProperty(
+				    HtmlFormEntryConstants.FORMATTER_DATETIME_NAME, "dd-MM-yyyy, HH:mm:ss"), Context.getLocale());
+				try {
+					String formatDateWithClientTZ = toClientTimezone(initialValue, Context.getAdministrationService()
+					        .getGlobalProperty(HtmlFormEntryConstants.FORMATTER_DATETIME_NAME));
+					if (StringUtils.isNotEmpty(formatDateWithClientTZ)) {
+						valAsCal.setTime(sdf.parse(formatDateWithClientTZ));
+					} else {
+						valAsCal.setTime(initialValue);
+					}
+				}
+				catch (ParseException e) {
```

Do you intentionally want to silently swallow this exception? Don't you want the client to some how know about it instead of just returning no data?

---

In that case the time will be 00:00:00, but I can send a message saying something like : "Error parsing the time, for encounter date." what do you think?

---

Displaying the error message is obviously much better than pretending all went well. 😊